piet/src/text.rs

```diff
@@ -102,8 +102,18 @@ impl Default for TextAlignment {
 ///
 pub trait TextLayout: Clone {
     /// Measure the advance width of the text.
+    #[deprecated(since = "0.2.0", note = "Use size().width insead")]
     fn width(&self) -> f64;
 
+    //TODO: ideally this would just be the bounding box of the glyphs? Otherwise
```

Comment:
Ultimately we will need *both* a layout size and a bounding box of the glyphs, for different reasons. Using one for the other is a persistent source of paper-cut bugs. I don't think this comment is very helpful as-is, as it seems to conflate the two concerns.

Regarding discrepancies between Core Text and DirectWrite, I think at some point we're going to have to do a careful review of vertical measurement, line space, and padding, something along the lines of [the Figma blog post on line height](https://www.figma.com/blog/line-height-changes/), and systematically document what we expect the behavior to be, as well as implement that behavior. It also needs to take into account scripts which have excessive line height (such as Myanmar). It's not a simple problem, as evidenced by continual wrestling over it.

Reply:
yea, left this comment in so we could have this discussion, and broadly I agree. I think the end game should be to get close to where figma is; in particular I think agreeing on a common definition of line-height would be really valuable, but that will require us to go to a runs-based API (or a line-based api that is backed by runs on most platforms).